includes/admin/plugins.php

```diff
@@ -129,108 +128,37 @@ function perflab_render_plugin_card( array $plugin_data ) {
-				);
-				if ( current_user_can( 'deactivate_plugin', $status['file'] ) ) {
-					$s       = isset( $_REQUEST['s'] ) ? $_REQUEST['s'] : ''; // phpcs:ignore WordPress.Security.NonceVerification.Recommended
-					$context = $status['status'];
-
-					$action_links[] = sprintf(
-						'<a href="%s" id="deactivate-%s" aria-label="%s">%s</a>',
-						esc_url(
-							add_query_arg(
-								array(
-									'_wpnonce' => wp_create_nonce( 'perflab_deactivate_plugin_' . $status['file'] ),
-									'action'   => 'perflab_deactivate_plugin',
-									'plugin'   => $status['file'],
-								),
-								network_admin_url( 'plugins.php' )
-							)
-						),
-						esc_attr( $plugin_data['slug'] ),
-						/* translators: %s: Plugin name. */
-						esc_attr( sprintf( _x( 'Deactivate %s', 'plugin', 'default' ), $plugin_data['slug'] ) ),
-						esc_html__( 'Deactivate', 'default' )
-					);
-				}
-			} elseif ( current_user_can( 'activate_plugin', $status['file'] ) ) {
-				if ( $compatible_php && $compatible_wp ) {
-					$button_text = __( 'Activate', 'default' );
-					/* translators: %s: Plugin name. */
-					$button_label = _x( 'Activate %s', 'plugin', 'default' );
-					$activate_url = add_query_arg(
-						array(
-							'_wpnonce' => wp_create_nonce( 'perflab_activate_plugin_' . $status['file'] ),
-							'action'   => 'perflab_activate_plugin',
-							'plugin'   => $status['file'],
-						),
-						network_admin_url( 'plugins.php' )
-					);
-
-					$action_links[] = sprintf(
-						'<a href="%1$s" class="button activate-now" aria-label="%2$s">%3$s</a>',
-						esc_url( $activate_url ),
-						esc_attr( sprintf( $button_label, $plugin_data['name'] ) ),
-						esc_html( $button_text )
-					);
-				} else {
-					$action_links[] = sprintf(
-						'<button type="button" class="button button-disabled" disabled="disabled">%s</button>',
-						esc_html( _x( 'Cannot Activate', 'plugin', 'default' ) )
-					);
-				}
-			} else {
-				$action_links[] = sprintf(
-					'<button type="button" class="button button-disabled" disabled="disabled">%s</button>',
-					esc_html( _x( 'Installed', 'plugin', 'default' ) )
-				);
-			}
-			break;
+	if ( is_plugin_active( $status['file'] ) ) {
+		$action_links[] = sprintf(
+			'<button type="button" class="button button-disabled" disabled="disabled">%s</button>',
+			esc_html__( 'Active', 'default' )
+		);
+	} elseif (
+		$compatible_php &&
+		$compatible_wp &&
+		current_user_can( 'install_plugins', $status['file'] ) &&
+		current_user_can( 'activate_plugin', $status['file'] )
+	) {
+		$action_links[] = sprintf(
+			'<button type="button" class="button perflab-install-active-plugin" data-slug="%s">%s</button>',
+			esc_attr( $plugin_data['slug'] ),
+			esc_html__( 'Activate', 'default' )
+		);
+	} else {
+		$action_links[] = sprintf(
+			'<button type="button" class="button button-disabled" disabled="disabled">%s</button>',
+			esc_html( _x( 'Cannot Install', 'plugin', 'default' ) )
+		);
```

Comment:
Let's correctly identify whether the plugin cannot be _activated_ or not be _installed_. Or alternatively, we always use "Cannot Activate" as that is more in line with the language on the active buttons.

Since this is a special case scenario, I'd say more granular explanations are still more helpful in conveying what's wrong.
```suggestion
		$explanation    = 'install' !== $status || current_user_can( 'install_plugins' ) ? _x( 'Cannot Activate', 'plugin', 'default' ) : _x( 'Cannot Install', 'plugin', 'default' );
		$action_links[] = sprintf(
			'<button type="button" class="button button-disabled" disabled="disabled">%s</button>',
			esc_html( $explanation )
		);
```